src/twisted/application/test/test_internet.py

```diff
@@ -829,3 +847,14 @@ def test_whenConnectedStopService(self):
         self.failureResultOf(a, CancelledError)
         self.failureResultOf(b, CancelledError)
 
+
+    def test_retryCancelled(self):
+        """
+        When L{ClientService.stopService} is called while waiting between connection attempts,
```

Comment:
Not enough space on the earlier docstrings, way too much on this one (95 characters)?